### DISCUSSION TITLE
osgEarth::GeometryUtils and OGR

### DISCUSSION THREAD

**Discussion by MCA4213** [pelicanmapping/osgearth#2455] [Support Q&A] [answered]
Hi, 
I am using osgEarth with spatial database to store geometries. For that I need to use WKT format to insert the geometry into the database.
1)	I used `osgEarth::GeometryUtils::geometryToWKT`  to do that, but this function generate a WKT that is not in iso format (generate `POINT` instead of `POINT Z` for vec3). 
As a suggestion, it is preferable to use `OGR_G_ExportToIsoWkt` instead of `OGR_G_ExportToWkt`, except if there is a purpose behind this implementation.
 
2)	The generated WKT is written in inverse order (order is important in my case). Take this example:
```
osgEarth::Geometry geom;
geom.push_back(1, 1, 0);
geom.push_back(2, 2, 0);
geom.push_back(3, 3, 0);
cout<< osgEarth::GeometryUtils::geometryToWKT(&geom);
```
RESULTS: `MULTIPOINT Z ((3 3 0),(2 2 0),(1 1 0))`
EXPECTED RESULTS: `MULTIPOINT Z ((1 1 0),(2 2 0),(3 3 0))`
The problem seems to be in OGRGeometryH OgrUtils::encodeShape that is adding points to OGR geometry from « geometry->size() – 1 » to 0 instead of starting from 0. 
Why?

Thanks.

**Comment by gwaldron** [comment]
Honestly, I do not have the faintest idea. :)
This code is so old I don't remember why reversing the order was necessary. If you want to try encoding in the same order, test it with some shapefiles (especially polygons) and we will take a PR for it.

**Comment by MCA4213** [comment, marked as answer]
Thanks, I have sent a PR. 
Also, I have tested loading shapefiles (point, line, polygon) and they are still loading correctly.
![image](https://github.com/gwaldron/osgearth/assets/7198569/277f0cb8-f1ca-4946-8931-ebf7f8f17702)
